internal/testrunner/runners/pipeline/test_case.go

```diff
@@ -0,0 +1,103 @@
+package pipeline
+
+import (
+	"bufio"
+	"bytes"
+	"encoding/json"
+	"regexp"
+	"strings"
+
+	"github.com/pkg/errors"
+)
+
+type testCase struct {
+	name   string
+	events []json.RawMessage
+}
+
+type testCaseDefinition struct {
+	Events []json.RawMessage `json:"events"`
+}
+
+type multiline struct {
+	FirstLinePattern string `json:"first_line_pattern"`
+}
+
+func createTestCaseForEvents(filename string, inputData []byte) (*testCase, error) {
+	var tcd testCaseDefinition
+	err := json.Unmarshal(inputData, &tcd)
+	if err != nil {
+		return nil, errors.Wrap(err, "unmarshalling input data failed")
+	}
+	return &testCase{
+		name:   filename,
+		events: tcd.Events,
+	}, nil
+}
+
+func createTestCaseForRawInput(filename string, inputData []byte, config testConfig) (*testCase, error) {
+	entries, err := readRawInputEntries(inputData, config)
+	if err != nil {
+		return nil, errors.Wrap(err, "reading raw input entries failed")
+	}
+
+	var events []json.RawMessage
+	for _, entry := range entries {
+		event := map[string]interface{}{}
+		event["message"] = entry
+
+		for k, v := range config.Fields {
+			event[k] = v
+		}
+
+		m, err := json.Marshal(&event)
+		if err != nil {
+			return nil, errors.Wrap(err, "marshalling mocked event failed")
+		}
+		events = append(events, m)
+	}
+	return &testCase{
+		name:   filename,
+		events: events,
+	}, nil
+}
+
+func readRawInputEntries(inputData []byte, c testConfig) ([]string, error) {
+	var inputDataEntries []string
+
+	var builder strings.Builder
+	scanner := bufio.NewScanner(bytes.NewReader(inputData))
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		var body string
+		if c.Multiline != nil && c.Multiline.FirstLinePattern != "" {
```

Comment:
Where does it exactly take this pattern from? How will it work with more complex patterns supported by Agent / Filebeat?

Reply:
I see, the pattern comes from the testcase directory config.